.github/workflows/flake8.yml

```diff
@@ -29,7 +29,8 @@ jobs:
         -   name: Setup python
             uses: actions/setup-python@v4
             with:
-                python-version: '3.10'
+                python-version: '3.11'
```

Comment:
I haven't looked closely at the flake8 failures related to 3.12 but at some point we'll need to handle these. Perhaps put in a separate issue

Reply:
agreed

they are not massive:

```
./examples/02_decoding/plot_haxby_grid_search.py:127:18: E226 missing whitespace around arithmetic operator
./nilearn/decoding/decoder.py:254:28: E201 whitespace after '{'
./nilearn/decoding/decoder.py:254:40: E231 missing whitespace after ':'
./nilearn/decoding/decoder.py:254:56: E202 whitespace before '}'
./nilearn/interfaces/fmriprep/tests/test_load_confounds.py:583:54: E226 missing whitespace around arithmetic operator
./nilearn/surface/surface.py:961:48: E201 whitespace after '{'
```

strangely though they are not "just" handled by black

Reply:
Ok maybe just manually handle here and update this workflow to 3.12?

Reply:
will try